cmd/agent/subcommands/streamlogs/command.go

```diff
@@ -72,8 +98,30 @@ func streamLogs(log log.Component, config config.Component, cliParams *cliParams
 	}
 
 	urlstr := fmt.Sprintf("https://%v:%v/agent/stream-logs", ipcAddress, config.GetInt("cmd_port"))
+
+	var f *os.File
+	var bufWriter *bufio.Writer
+
+	if cliParams.FilePath != "" {
+		f, bufWriter, err = openFileForWriting(cliParams.FilePath)
+		if err != nil {
+			return fmt.Errorf("error opening file %s for writing: %v", cliParams.FilePath, err)
+		}
+		defer f.Close()
+
+		if bufWriter != nil {
+			defer bufWriter.Flush()
+		}
+	}
+
 	return streamRequest(urlstr, body, func(chunk []byte) {
 		fmt.Print(string(chunk))
+
+		if bufWriter != nil {
+			if err = writeToBuffer(bufWriter, string(chunk)); err != nil {
```

Comment:
NIT: In this case, i'd just call `bufWriter.WriteString(message)` here and remove the helper (`writeToBuffer`) - since it just stacks error handling. The error handling can be easily combined into one line (below - where you `Printf`). Will save a few LOC as well.